apps/emqx_rule_engine/src/emqx_rule_engine_jwt_worker.erl

```diff
@@ -0,0 +1,220 @@
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%--------------------------------------------------------------------
+
+-module(emqx_rule_engine_jwt_worker).
+
+-behaviour(gen_server).
+
+%% API
+-export([ start_link/2
+        , lookup_jwt/1
+        , lookup_jwt/2
+        ]).
+
+%% gen_server API
+-export([ init/1
+        , handle_continue/2
+        , handle_call/3
+        , handle_cast/2
+        , handle_info/2
+        , format_status/1
+        , format_status/2
+        ]).
+
+-include_lib("jose/include/jose_jwk.hrl").
+-include_lib("emqx_rule_engine/include/rule_engine.hrl").
+-include_lib("emqx_rule_engine/include/rule_actions.hrl").
+-include_lib("snabbkaffe/include/snabbkaffe.hrl").
+
+-type config() :: #{ private_key := binary()
+                   , resource_id := resource_id()
+                   , expiration := timer:time()
+                   , table := ets:table()
+                   , iss := binary()
+                   , sub := binary()
+                   , aud := binary()
+                   , kid := binary()
+                   , alg := binary()
+                   }.
+-type jwt() :: binary().
+-type state() :: #{ refresh_timer := undefined | timer:tref()
+                  , resource_id := resource_id()
+                  , expiration := timer:time()
+                  , table := ets:table()
+                  , jwt := undefined | jwt()
+                    %% only undefined during startup
+                  , jwk := undefined | jose_jwk:key()
+                  , iss := binary()
+                  , sub := binary()
+                  , aud := binary()
+                  , kid := binary()
+                  , alg := binary()
+                  }.
+
+-define(refresh_jwt, refresh_jwt).
+
+%%-----------------------------------------------------------------------------------------
+%% API
+%%-----------------------------------------------------------------------------------------
+
+-spec start_link(config(), reference()) -> gen_server:start_ret().
+start_link(#{ private_key := _
+            , expiration := _
+            , resource_id := _
+            , table := _
+            , iss := _
+            , sub := _
+            , aud := _
+            , kid := _
+            , alg := _
+            } = Config,
+           Ref) ->
+    gen_server:start_link(?MODULE, {Config, Ref}, []).
+
+-spec lookup_jwt(resource_id()) -> {ok, jwt()} | {error, not_found}.
+lookup_jwt(ResourceId) ->
+    ?MODULE:lookup_jwt(?JWT_TABLE, ResourceId).
+
+-spec lookup_jwt(ets:table(), resource_id()) -> {ok, jwt()} | {error, not_found}.
+lookup_jwt(TId, ResourceId) ->
```

Comment:
it's odd that table is owned by the caller, but yet accessed from here.
possible to make the supervisor own this table and share the usage for all children (jwt worker processes).

Reply:
The table creation is not done in this PR, but it'll be done in the [rule action](https://github.com/emqx/emqx-enterprise/blob/ce8ede34f611cbf3285b82d78bb1c69b9878e956/lib-ee/emqx_rule_actions/src/emqx_backend_gcp_pubsub_actions.erl#L654-L667).

In practice, the supervisor will be the owner of the table for all workers / rule actions.

Reply:
Why not just create it by the supervisor?

Reply:
Well, I thought this would ensure it's always there, regardless of anything.

I also found [something similar](https://github.com/emqx/emqx-enterprise/blob/9b038563b36ded0a1b1d1796cde06c9fb0dec0f9/lib-ee/emqx_rule_actions/src/emqx_backend_sap_event_mesh.erl#L404-L413) in another action, so I thought this was the way.

I'll move it to the supervisor then.

Reply:
:heavy_check_mark: